lib/sprite_batch.dart

```diff
@@ -1,67 +1,267 @@
+      defaultColor: defaultColor,
+      defaultTransform: defaultTransform ?? RSTransform(1, 0, 0, 0),
+      defaultBlendMode: defaultBlendMode,
+    );
+  }
+
+  /// Add a new batch item using a RSTransform.
+  ///
+  /// The [source] parameter is the source location on the [atlas]. You can position it
+  /// on the canvas using the [offset] parameter.
+  ///
+  /// The [color] paramater allows you to render a color behind the batch item, as a background color.
+  ///
+  /// The [add] method may be a simpler way to add a batch item to the batch. However,
+  /// if there is a way to factor out the computations of the sine and cosine of the
+  /// rotation so that they can be reused over multiple calls to this constructor,
+  /// it may be more efficient to directly use this method instead.
   void addTransform({
-    @required Rect rect,
+    @required Rect source,
     RSTransform transform,
     Color color,
   }) {
-    rects.add(rect);
-    transforms.add(transform ?? defaultTransform);
-    colors.add(color ?? defaultColor);
+    final batchItem = BatchItem(
+      source: source,
+      transform: transform ??= defaultTransform ?? RSTransform(1, 0, 0, 0),
+      color: color ?? defaultColor,
+    );
+
+    _batchItems.add(batchItem);
+
+    _sources.add(batchItem.source);
+    _transforms.add(batchItem.transform);
+    _colors.add(batchItem.color);
   }
 
+  /// Add a new batch item.
+  ///
+  /// The [source] parameter is the source location on the [atlas]. You can position it
+  /// on the canvas using the [offset] parameter.
+  ///
+  /// You can transform the sprite from its [offset] using [scale], [rotation] and [anchor].
+  ///
+  /// The [color] paramater allows you to render a color behind the batch item, as a background color.
+  ///
+  /// This method creates a new [RSTransform] based on the given transform arguments. If many [RSTransform] objects are being
+  /// created and there is a way to factor out the computations of the sine and cosine of the rotation
+  /// (which are computed each time this method is called) and reuse them over multiple [RSTransform] objects,
+  /// it may be more efficient to directly use the more direct [addTransform] method instead.
   void add({
-    @required Rect rect,
+    @required Rect source,
     double scale = 1.0,
     Offset anchor = Offset.zero,
     double rotation = 0,
     Offset offset = Offset.zero,
     Color color,
   }) {
-    final transform = RSTransform.fromComponents(
-      scale: scale,
-      anchorX: anchor.dx,
-      anchorY: anchor.dy,
-      rotation: rotation,
-      translateX: offset.dx,
-      translateY: offset.dy,
-    );
-    addTransform(rect: rect, transform: transform, color: color);
+    RSTransform transform;
+
+    // If any of the transform arguments is different from the defaults,
+    // then we create one. This is to prevent unnecessary computations
+    // of the sine and cosine of the rotation.
+    if (scale != 1.0 ||
```

Comment:
This will add an empty transform when this if is false, is that desired?

Reply:
It calls the `addTransform` and the `addTransform` already checks if the given transform(which is optional) is null, if so it will set a default one that does not do any calculations.